samples/Monitoring/deploy-mmaextension-windows-tags-loganalytics/azurepolicy.json

```diff
@@ -0,0 +1,140 @@
+                                "equals": "Succeeded"
+                            }
+                        ]
+                    },
+                    "deployment": {
+                        "properties": {
+                            "mode": "incremental",
+                            "template": {
+                                "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                                "contentVersion": "1.0.0.0",
+                                "parameters": {
+                                    "vmName": {
+                                        "type": "string"
+                                    },
+                                    "location": {
+                                        "type": "string"
+                                    },
+                                    "logAnalytics": {
+                                        "type": "string"
+                                    },
+                                    "workspaceId": {
+                                        "type": "string"
+                                    }
+                                },
+                                "variables": {
+                                    "vmExtensionName": "MicrosoftMonitoringAgent",
+                                    "vmExtensionPublisher": "Microsoft.EnterpriseCloud.Monitoring",
+                                    "vmExtensionType": "MicrosoftMonitoringAgent",
+                                    "vmExtensionTypeHandlerVersion": "1.0"
+                                },
+                                "resources": [{
+                                    "name": "[concat(parameters('vmName'), '/', variables('vmExtensionName'))]",
+                                    "type": "Microsoft.Compute/virtualMachines/extensions",
+                                    "location": "[parameters('location')]",
+                                    "apiVersion": "2018-06-01",
+                                    "properties": {
+                                        "publisher": "[variables('vmExtensionPublisher')]",
+                                        "type": "[variables('vmExtensionType')]",
+                                        "typeHandlerVersion": "[variables('vmExtensionTypeHandlerVersion')]",
+                                        "autoUpgradeMinorVersion": true,
+                                        "settings": {
+                                            "workspaceId": "[reference(parameters('logAnalytics'), '2015-03-20').customerId]",
+                                            "stopOnMultipleConnections": "true"
+                                        },
+                                        "protectedSettings": {
+                                            "workspaceKey": "[listKeys(resourceId('Microsoft.OperationalInsights/workspaces', parameters('loganalytics')), '2015-03-20').primarySharedKey]"
```

Comment:
>listKeys(resourceId('Microsoft.OperationalInsights/workspaces', parameters('loganalytics') [](start = 62, length = 90)

this also needs to be updated

Reply:
I have tested this locally and it works as expected. I am using resourceId instead of reference here and i've tested it :
![image](https://user-images.githubusercontent.com/22794646/63453506-70d18b80-c466-11e9-8d7c-8881737d38ad.png)

Reply:
This will be a runtime problem during template deployment execution, not a problem with the policy itself.  the resourceId() function takes type, name, etc... separately.  The "loganalytics" parameter is the full log analytics workspace resource ID.  You should be using it here very similarly to how it was used in the "reference()" function above.

[https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#resourceid](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#resourceid)

---
In reply to: [316305304](https://github.com/Azure/azure-policy/pull/435#discussion_r316305304) [](ancestors = 316305304)

Reply:
> This will be a runtime problem during template deployment execution, not a problem with the policy itself. the resourceId() function takes type, name, etc... separately. The "loganalytics" parameter is the full log analytics workspace resource ID. You should be using it here very similarly to how it was used in the "reference()" function above.
> 
> https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-resource#resourceid
> 
> In reply to: [316305304](https://github.com/Azure/azure-policy/pull/435#discussion_r316305304) [](ancestors = 316305304)

I have made the changes - good call outs @pilor

Reply:
The deployment in your tags policy will still fail.  There are parameters in the deployment that aren't provided, the resourceID function in listKeys is invalid, etc...